src/collection/element.js

```diff
@@ -87,6 +87,14 @@ let Element = function( cy, params, restore ){
 
       _p.classes.add(cls);
     }
+  } else if( is.array( params.classes ) ){
```

Comment:
Doing an "if/else if" where both blocks only differ by one line feels messy. I have a few refactors in mind, but I held off because I wanted to preserve as much of the original code as possible for my initial PR. Also, this is arguably simple/small enough that refactoring might not make a huge difference anyway.

Reply:
Since the code is the same apart from assigning classes, you could remove the if/else and just use a ternary operator (`?`) inline in the declaration of `classes`.

Reply:
If I understand correctly, doing that would make it something like
```
if ( is.array( params.classes ) || is.string( params.classes ) ){
  let classes = is.string( params.classes ) ? params.classes.split( /\s+/ ) : params.classes;
  // ...
```
which still feels kind of wonky to me. One of the refactors I had in mind was pulling the for loop out of the conditional and having the conditional just set the value for the `classes` variable. If params.classes is not a string or array, it'll just default to being an empty array (and therefore would be skipped by the for loop without causing errors.)

Maybe something like
```
let classes = params.classes || [];
if ( is.string( classes ) ){
  classes = classes.split( /\s+/ );
}
for( let i = 0, l = classes.length; i < l; i++ ){
  let cls = classes[ i ];
  if( !cls || cls === '' ){ continue; }

  _p.classes.add(cls);
}
```
This is probably the cleanest option I have in mind, but it now will throw an error when you put an invalid type in for `classes` instead of silently passing through, which might break some people's code. To avoid that, I'm thinking something like
```
let classes = [];
if ( is.array( params.classes ) ){
  classes = params.classes;
} else if ( is.string( params.classes ) ){
  classes = params.classes.split( /\s+/ );
}
for( let i = 0, l = classes.length; i < l; i++ ){
  let cls = classes[ i ];
  if( !cls || cls === '' ){ continue; }

  _p.classes.add(cls);
}
```
It's not as clean as the other one, but it won't run the risk of breaking people's stuff. On the other hand, one could argue that anything that depends on passing invalid data in for `classes` is stuff that should break anyway.

At the end of the day, it's a matter maintainer preference. I'm fine with any of these options (including the option of just leaving it as is and calling it a day.)

Reply:
I was thinking of the third one basically:

```js
let classes = is.array(params.classes) ? params.classes : is.string(params.classes) ? params.classes.split( /\s+/ ) : null;
if( classes != null ){ for( let i = 0, l = classes.length; i < l; i++ ){
  let cls = classes[ i ];
  if( !cls || cls === '' ){ continue; }

  _p.classes.add(cls);
} }
```